crates/gossip/src/state.rs

```diff
@@ -0,0 +1,194 @@
+use crate::protocol::{
+    K2GossipAcceptMessage, K2GossipAgentsMessage, K2GossipNoDiffMessage,
+};
+use kitsune2_api::{AgentId, K2Error, K2Result, Url};
+use rand::RngCore;
+
+/// The state of a gossip round.
+#[derive(Debug)]
+pub(crate) struct GossipRoundState {
```

Comment:
This is a similar thing to what we had in K1. It's just about keeping track of the gossip round so that we can apply a timeout and check that the other peer is sending us messages in the sequence that we're expecting, based on our state as we go.